src/main/java/org/spongepowered/api/service/context/Context.java

```diff
@@ -44,18 +45,32 @@
     public static final String LOCAL_IP_KEY = "localip";
     public static final String LOCAL_PORT_KEY = "localport"; // This portkey is an old boot
 
-    private final Map.Entry<String, String> wrapped;
+    private final String key;
+    private final String value;
+    private final int weight;
 
     /**
      * Create a new context instance
      *
      * @param type Context type. Must not be null.
      * @param name Context name. Must not be null.
+     * @param weight Context weight. Must be greater than or equal to 1. Default is 1.
+     */
+    public Context(String type, String name, int weight) {
+        this.key = checkNotNull(type, "type");
+        this.value = checkNotNull(name, "name");
+        checkArgument(weight >= 1, "weight must be >= 1");
```

Comment:
Why is there such a limitation? What if I want to have my context less than default weight?